r/R/dataset.R

```diff
@@ -145,29 +145,24 @@ open_dataset <- function(sources,
 #' @seealso [open_dataset()] for a simple interface to creating a `Dataset`
 Dataset <- R6Class("Dataset", inherit = ArrowObject,
   public = list(
-    ..dispatch = function() {
-      type <- self$type
-      if (type == "union") {
-        shared_ptr(UnionDataset, self$pointer())
-      } else if (type == "filesystem") {
-        shared_ptr(FileSystemDataset, self$pointer())
-      } else {
-        self
-      }
-    },
     # @description
     # Start a new scan of the data
     # @return A [ScannerBuilder]
-    NewScan = function() unique_ptr(ScannerBuilder, dataset___Dataset__NewScan(self)),
-    ToString = function() self$schema$ToString()
+    NewScan = function() dataset___Dataset__NewScan(self),
+    ToString = function() self$schema$ToString(),
+    write = function(path, filesystem = NULL, schema = self$schema, format, partitioning, ...) {
+      path_and_fs <- get_path_and_filesystem(path, filesystem)
+      dataset___Dataset__Write(self, schema, format, path_and_fs$fs, path_and_fs$path, partitioning)
+      invisible(self)
+    }
```

Comment:
Another removal from the latest dataset writing patch

```suggestion
    ToString = function() self$schema$ToString()
```